src/include/optimizer/optimizer_task_pool.h

```diff
@@ -0,0 +1,53 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// rule.h
+//
+// Identification: src/optimizer/optimizer_task_pool.cpp
```

Comment:
Since this is really a stack, I think it's probably better to name it optimizer_task_stack. Unless you have some reason to call it a pool?

Reply:
For single-threaded optimization, this should be a stack. In the future, if we are going to implement a multi-threaded optimizer, the task pool may be implemented using other data structures.

Reply:
I see. Interesting. I think it's worth to note that in the code as well.